### ISSUE TITLE
App crash completely after update to Meteor 1.3

### ISSUE THREAD

**Issue by cyclops24** [perak/kitchen-site#323]
Hi @perak ,
I have a app based on meteor-kitchen on Meteor 1.2.1, Now I update my project to 1.3.4.4 and app crash completely.
I can run my project successfully in Meteor 1.2.1 but now after update My app crashed with iron-router message "Organize your Meteor application" like below image:

![meteor 13 issues 2](https://cloud.githubusercontent.com/assets/8266479/16893230/b2e3b54e-4b46-11e6-9aa5-e32457a0caca.png)

Why this error happend? I used Iron-Router in all of my project parts. How to fix this?

**Comment by perak** [comment]
@cyclops24 no problems with 1.3.4.4 here. Must be something specific in your code.

BTW, does this happens if you generate any application ( example-minimal ) too, or happens only with this project you mentioned?

**Comment by perak** [comment]
Can you show me list of packages used in the project?

**Comment by cyclops24** [comment]
I also see below warning after run `meteor update`:
```
Unable to resolve some modules:

  "jquery" in /home/user/Programming/IProjects/tttest/Sources/client/lib/select2/select2.full.js
(web.browser)

If you notice problems related to these missing modules, consider running:

  meteor npm install --save jquery
```

This is my `meteor list` output:
```
accounts-base                 1.2.8  A user account system
accounts-password             1.1.12  Password support for accounts
alanning:roles                1.2.15  Authorization package for Meteor
aldeed:autoform               5.8.1  Easily create forms with automatic insert and update, and automatic reactive validation.
aldeed:autoform-select2       2.0.2  Custom select2 input type for AutoForm
aldeed:collection2            2.9.1  Automatic validation of insert and update operations on the client and server.
aldeed:schema-index           1.0.1  Control some MongoDB indexing with schema options
aldeed:simple-schema          1.5.3  A simple schema validation object with reactivity. Used by collection2 and autoform.
bigdsk:inputmask              3.1.63  JQuery InputMask for Meteor
blaze-html-templates          1.0.4  Compile HTML templates into reactive UI with Meteor Blaze
check                         1.2.3  Check whether a value matches a pattern
copleykj:livestamp            1.1.7  jQuery plugin providing auto-updating timeago text to your timestamped HTML elements using Moment.js
ecmascript                    0.4.7  Compiler plugin that supports ES2015+ in all .js files
ejson                         1.0.12  Extended and Extensible JSON library
email                         1.0.15  Send email messages
es5-shim                      4.5.13  Shims and polyfills to improve ECMAScript 5 support
http                          1.1.8  Make HTTP calls to remote servers
iron:router                   1.0.13  Routing specifically designed for Meteor
jquery                        1.11.9  Manipulate the DOM using CSS selectors
less                          2.6.4  Leaner CSS language
matb33:collection-hooks       0.8.1  Extends Mongo.Collection with before/after hooks for insert/update/remove/find/findOne
maximum:server-transform      0.5.0  Meteor package to transform documents on publish
meteor-base                   1.0.4  Packages that every Meteor app needs
mizzao:bootboxjs              4.4.0  Programmatic dialog boxes using Twitter's bootstrap modals
mobile-experience             1.0.4  Packages for a great mobile user experience
mongo                         1.1.9_1  Adaptor for using MongoDB and Minimongo over DDP
mrt:moment                    2.8.1  Moment.js, a JavaScript date library for dates, packaged for Meteor. See http://momentjs.com.
natestrauser:font-awesome     4.6.3  Latest version Font-Awesome loaded via CDN
nemo64:bootstrap              3.3.5_3  Highly configurable bootstrap integration.
perak:joins                   1.0.6  Generic joins for Meteor
perak:user-roles              1.0.10  User roles package for Meteor
rajit:bootstrap3-datepicker   1.5.1  Meteor packaging of eternicode/bootstrap-datepicker for Bootstrap 3
random                        1.0.10  Random number generator and utilities
reactive-dict                 1.1.8  Reactive dictionary
session                       1.1.6  Session variable
spiderable                    1.0.13  Makes the application crawlable to web spiders
standard-minifier-css         1.0.8  Standard css minifier used with Meteor apps by default.
standard-minifier-js          1.0.8  Standard javascript minifiers used with Meteor apps by default.
tracker                       1.0.14  Dependency tracker to allow reactive callbacks
zimme:select2-bootstrap3-css  1.4.6  CSS to make Select2 fit in with Bootstrap 3
```

I try to generate `example-minimal` and send result here.

**Comment by perak** [comment]
Did you run meteor npm install ... as suggested in error message?

**Comment by cyclops24** [comment]
I can generate and run `example-minimal` successfully Meteor Kitchen v0.9.57.
@perak I create this project near 6 month ago. Is some important part change in `router.js` file that related to this issue??

**Comment by cyclops24** [comment]
I can generate and run `example-minimal` successfully Meteor Kitchen v0.9.57.
@perak I create this project near 6 month ago. Is some important part change in `router.js` file that related to this issue??

**Comment by perak** [comment]
No significant changes in router.js (as far as I remember)

**Comment by cyclops24** [comment]
I also try install jquery but still `Organize your Meteor application` error :worried:

**Comment by cyclops24** [comment]
@perak I also try to create new project and copy sources from my project to this new one but still not wroking....

**Comment by perak** [comment]
@cyclops24 what happened? Did you find the bug.

Maybe I can help, but i'll need to see your code.

**Comment by cyclops24** [comment]
Hi @perak,
I also ask this on [meteor forum](https://forums.meteor.com/t/some-problem-for-update-one-project-from-1-2-1-to-1-3-4-4/26437/3)
My project is a little big which part of code you need to see man?
It's very amazing for me that app running smoothly on 1.2.1 but not work on 1.3.x at all.
I also update from 1.3.4.4 to 1.3.5.1 but still route error exist.

**Comment by perak** [comment]
Hi @cyclops24 

Did you try:

- clean build (generate into new empty dir)

- update meteor-kitchen to the latest and clean build again

**Comment by perak** [comment]
Also try:

```
meteor update iron:middleware-stack
```

**Comment by cyclops24** [comment]
Can you explain `clean build`?

**Comment by perak** [comment]
If you are using kitchen, build into new (non-existing) directory instead overwriting existing one.

**Comment by cyclops24** [comment]
@perak I change a lot of files in generated code but this project is based on meteor kitchen. (and also my router)
I search a lot and another user have same issue with iron router here:
https://github.com/iron-meteor/iron-router/issues/1522

**Comment by perak** [comment]
@cyclops24 I had the same issue yesterday when I made a bug (client side) after which rest of code stopped executing. Do you have any bugs in browser console?

**Comment by cyclops24** [comment]
Hi man,
Yes I have below error on client but I can't debug and find where is it.
![meteor13error](https://cloud.githubusercontent.com/assets/8266479/17088646/2a47356e-5233-11e6-8bcb-f30c849bb592.png)
And without any error on server console.
I can't understand where is `app.js` file and how to debug this to find issue.

**Comment by perak** [comment]
@cyclops24 what happens if you click at the filename/linenumber at the right in the same line where is error message? Would browser open source file and position at the line where error occours? If not, try the same with Chrome. And let me see the problematic code snippet.

**Comment by cyclops24** [comment]
Hi @perak,
This is code snippet when click on filename/linenumber:
![meteor13issue_2](https://cloud.githubusercontent.com/assets/8266479/17100878/7a08b0ac-5286-11e6-8a2e-0f130c02fc99.png)

@perak you save me man. You point to Chrome and this note save my life. I saw that project not working on Chrome too but I didn't see console output on Chrome. This is error message on Chrome:
![screenshot from 2016-07-25 16 42 56](https://cloud.githubusercontent.com/assets/8266479/17101004/3964de12-5287-11e6-9048-043e6a84cfcf.png)

I don't know why but on Chrome more error printed on console and the main error is `ar.js:3Uncaught TypeError: Cannot read property 'define' of undefined` and this is a `i18n` files of `select2` component that I used in project in `client/lib/select2/i18n` path. I remove this folder at all and now project works well without any issue. :smiley:
